cmd/osbuild-worker/jobimpl-depsolve.go

```diff
@@ -42,6 +44,22 @@ func (impl *DepsolveJobImpl) Run(job worker.Job) error {
 		return err
 	}
 
+	if impl.RepoMTLS != nil {
+		for _, pkgsets := range args.PackageSets {
+			for _, pkgset := range pkgsets {
+				for _, repo := range pkgset.Repositories {
+					for _, baseurl := range repo.BaseURLs {
+						if strings.HasPrefix(baseurl, impl.RepoMTLS.BaseURL) {
```

Comment:
I would really prefer if we did the `BaseURL` stuff by actually parsing URLs.

Reply:
So do you imagine comparing the hostname, the scheme, the path separately? Or just to parse to make sure it's a valid url?

Reply:
```
	if impl.RepositoryMTLSConfig != nil {
		for _, pkgsets := range args.PackageSets {
			for _, pkgset := range pkgsets {
				for _, repo := range pkgset.Repositories {
					for _, baseurlstr := range repo.BaseURLs {
						baseurl, err := url.Parse(baseurlstr)
						if err != nil {
							result.JobError = clienterrors.WorkerClientError(clienterrors.ErrorInvalidRepositoryURL)
							return err
						}
						if baseurl.Host = impl.RepositoryMTLSConfig.BaseURL.Host && strings.HasPrefix(baseurl.Path, impl.RepositoryMTLSConfig.BaseURL.Path) {
							repo.SSLCACert = impl.RepositoryMTLSConfig.CA
							repo.SSLClientKey = impl.RepositoryMTLSConfig.MTLSClientKey
							repo.SSLClientCert = impl.RepositoryMTLSConfig.MTLSClientCert
						}
					}
				}
			}
		}
	}

``` 

something like this?

Reply:
Yes, exactly like that :)